### ISSUE TITLE
Ideas for providing more control over column sorting

### ISSUE THREAD

**Issue by seanthebean** [handsontable/handsontable#2880]
I'm working on a project that requires column sorting, and I ran into a couple short-comings of the current functionality:

1. I have not found any way to specify a custom comparator function to use when sorting individual columns. Specifically, we need to be able to use a natural sorting algorithm (https://en.wikipedia.org/wiki/Natural_sort_order), specifically http://phpjs.org/functions/strnatcasecmp/, for text-based columns (which most of the columns in the table are). Ideally, I think it should be possible to specify the comparator at the column level (i.e., in the column options), so columns could potentially be sorting using different comparators that are appropriate to each (e.g. we wouldn't use the natsort comparator for a boolean/checkbox column).

2. I was surprised that the sort column can only be specified by assigning a number to the `column` property of the `configObject`, but not a string corresponding to the property name. It would be great to be able to use the property name in cases where the column numbers may change (i.e., if adding, removing, or re-ordering columns). The project I'm working on requires dynamically hiding and showing sets of columns.

3. I was also surprised that it does not seem to be possible to sort by multiple columns in order of preference. For instance, you might have a Last Name column and a First Name column. It would make sense to sort the data alphabetically by Last Name, using the First Name to discriminate between rows that have the same Last Name. This could be accomplished by allowing an array of `configObject`s to be assigned to the `columnSorting` property.

4. It might also be useful (as an alternative or in addition to the above) to sort the data just using a custom function. To support this functionality, the `columnSorting` property could accept a comparator function as well.

I am very interested in seeing any or all of these implemented. (Note: I can split this into 4 different issues if that would be easier to manage, but I thought it might be useful to see them all in once place first, in case they would conflict with each other.)

**Comment by mamapitufo** [comment]
Any guess on when this functionality could be available? Would it be a good idea to submit a PR for at least some of the features?

**Comment by seanthebean** [comment]
Sure!

And just in case anyone else is in a similar situation, I ended up working around these issues by turning off `columnSorting` and sorting the source data instead. But I think the new functionality is still highly desirable, because sorting the source data is not ideal (may cause issues if other parts of your code depend on the order of the array), and this prevents having the `sortIndicator` in the column header, unless we go to the work of re-implementing that in our application as well.

**Comment by grandtiger** [comment]
I really need sorting by multiple columns as well. handsontable seems to have a lot of good features, but the lack of multi-column sorting and grouping features makes it not suitable for a lot of serious projects.

**Comment by AMBudnik** [comment]
Thank you all for feedback and support - we already reported this feature at our feathub at http://feathub.com/handsontable/handsontable/+5 although first things comes first. We have to fix some bugs before we can go further and implement new features.
